src/cleanfigure.m

```diff
@@ -939,6 +939,30 @@ function removeNaNs(meta, handle)
     topRight    = [xLim(2); yLim(2)];
 end
 % =========================================================================
+function data = limitPrecision(data)
+  % Limit the precision of the given data
+  
+  % Only do something if the data is not empty
+  if isempty(data) || all(~isfinite(data))
+      return
```

Comment:
Actually this captures two different problems. The first is an empty range. The second is that range or maxValue would be NaN if one of the data points is inf or NaN.